x-pack/plugin/sql/qa/server/src/main/java/org/elasticsearch/xpack/sql/qa/rest/RestSqlTestCase.java

```diff
@@ -1438,6 +1444,126 @@ public void testCompressCursor() throws IOException {
         assertThat(resp.get("cursor").toString().length(), lessThan(5000));
     }
 
+    public void testFetchAllPagesSearchHitCursorTxt() throws IOException {
```

Comment:
@astefan I've extended the IT a bit to ensure that we cover all combinations of (cursorType, format) together with random sizes and page sizes. Altogether it should cover all scenarios including "all records in first page", "full last page", "empty last page" etc.